pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationProvider.java

```diff
@@ -345,9 +362,41 @@ default CompletableFuture<Boolean> allowTopicOperationAsync(TopicName topic,
                                                                 String role,
                                                                 TopicOperation operation,
                                                                 AuthenticationDataSource authData) {
-        return FutureUtil.failedFuture(
-            new IllegalStateException("TopicOperation [" + operation.name() + "] is not supported by the Authorization"
-                    + "provider you are using."));
```

Comment:
Are you able to explain a bit more why this is required for backwards compatibility? I think we should be able to leave this here and leave the `switch` statement in the `PulsarAuthorizationProvider`.